search/services/index/prepare.py

```diff
@@ -204,8 +213,65 @@ def _query_all_fields(term: str) -> Q:
         _query_primary(term, operator='or')
     ]
 
-    if date_partial:
-        queries.insert(0, Q("term", announced_date_first=date_partial))
+    # It is possible that the query includes a date-related term, which we
+    # interpret as an announcement date of v1 of the paper. We currently
+    # support both "standard" `yyyy` or `yyyy-MM`` formats as well as a
+    # legacy format ``yyMM``.
+    #
+    # The general strategy here is to first attempt to match a date fragment
+    # using one the formats above, and split the query so that we can handle
+    # the date fragment and the remainder of the query separately. If we find
+    # something that looks like a date fragment, we perform the all-fields
+    # search on the remainder and use the fragment to build queries against the
+    # announcement-date of the original paper version.
+    date_fragment: Optional[str] = None
+    remainder: Optional[str] = None
+    try:
+        date_fragment, remainder = parse_date(term)
+    except ValueError:
+        pass
+
+    if date_fragment:
+        logger.debug('date: %s; remainder: %s', date_fragment, remainder)
+        match_date: Optional[Q] = None
+        match_date_partial: Optional[Q] = None
+        match_date_announced: Optional[Q] = None
+        match_dates: List[Q] = []
+        logger.debug('date_fragment: %s', date_fragment)
+
+        # Try to query using legacy yyMM date partial format.
```

Comment:
@mhl10 Should this be part of the feature that implements ARXIVNG-983 (support searches for seven-digit arXiv ID partials)?